From 02095fb5e081454b72d8343b357669a75588eaa6 Mon Sep 17 00:00:00 2001
From: Matthew White <mehw@users.noreply.github.com>
Date: Mon, 12 Oct 2026 08:42:14 +0000
Subject: [PATCH] Fix implicit declaration of functions explicit_bzero and
 explicit_memset by adding the <string.h> header (#620)

* Fix implicit declaration of function explicit_bzero

This completes commit c3ba909e296e824f794e26d36379174524af74b8.

Add missing string.h header, required to define explicit_bzero.

* Fix implicit declaration of function explicit_memset

This completes commit c3ba909e296e824f794e26d36379174524af74b8.

Add missing string.h header, required to define explicit_memset.
---
 lib/zipint.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/zipint.h b/lib/zipint.h
index 25abcb13..5de9052f 100644
--- a/lib/zipint.h
+++ b/lib/zipint.h
@@ -524,9 +524,11 @@ typedef struct _zip_pkware_keys zip_pkware_keys_t;
 
 
 #ifdef HAVE_EXPLICIT_MEMSET
+#include <string.h>
 #define _zip_crypto_clear(b, l) explicit_memset((b), 0, (l))
 #else
 #ifdef HAVE_EXPLICIT_BZERO
+#include <string.h>
 #define _zip_crypto_clear(b, l) explicit_bzero((b), (l))
 #else
 #define USE_OWN_CRYPTO_CLEAR
-- 
2.56.0

